page_group.php

```diff
@@ -0,0 +1,145 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Page group meta data.
+ *
+ * @package   tool_excimer
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @copyright 2022, Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+use core\chart_line;
+use core\chart_series;
+use tool_excimer\helper;
+use tool_excimer\output\tabs;
+use tool_excimer\page_group;
+
+require_once('../../../config.php');
+require_once($CFG->libdir.'/adminlib.php');
+
+$expiry = get_config('tool_excimer', 'expiry_fuzzy_counts');
+$pagegroupid = required_param('id', PARAM_INT);
+$monthstodisplay = optional_param('monthstodisplay', $expiry, PARAM_INT);
+
+$url = new \moodle_url('/admin/tool/excimer/page_group.php');
+$context = context_system::instance();
+
+$pagegroup = new page_group($pagegroupid);
+
+$PAGE->set_context($context);
+$PAGE->set_url($url);
+$PAGE->navbar->add($pagegroup->get('name'));
+
+admin_externalpage_setup('tool_excimer_report_page_groups');
+
+$output = $PAGE->get_renderer('tool_excimer');
+
+$data = $pagegroup->to_record();
+
+// Data for table.
+$data->month = userdate(strtotime($data->month . '01'), '%b %Y');
```

Comment:
There does not seem to be an equivalent format as this one, but a similar one exists as `'%B %Y'`

Which would mean the display changes from `Sep 2022` -> `September 2022`. That said I'm a bit reluctant to change it to the long form. It is being used also on the page_group_table class. Could we move this format to a `const` and reference it in both places for consistency?

The alternative would be to move this into a helper function (which might happen for something similar below), perhaps something such as `get_month_formatted(int $timestamp)`

Reply:
The format itself is now part of the string table, and the whole thing moved to a helper function.